rfcs/block-component.md

```diff
@@ -0,0 +1,78 @@
+# Block Component
+
+One of the most common needs when building web apps is creating layout elements that have certain layout and typography styles. 
+
+For more traditional apps that use global stylesheets, many of the popular CSS frameworks have helper classes to quickly create styled elements (see [Bootstrap](https://getbootstrap.com/docs/4.1/utilities/sizing/), [Tachyons](http://tachyons.io/docs/layout/spacing/)). Uber's superfine css framework also had this and it was used very widely–just search something like `push-large` in sourcegraph.
+
+The move to CSS-in-JS sort of eliminates these classes in favor of encapsulating styles within components. This is a generally a good thing because it promotes reusability, however the reality is that there are dozens or hundreds of different layout/spacing/typography combinations in any app, and there's not always a clear semantic meaning or potential for reuse. In practice you end up just defining and naming a bunch of things at the top of the file:
+
+```js
+const PricingPageHero = styled('div', {
+  margin: '10px 15px',
+  color: '#ccc',
+  display: 'flex',
+});
+
+const PricingPageHeroText = styled('h2', {
+  margin: '10px 0',
+});
+
+// ... and many more
+```
+
+---
+
+## Proposal
+
+### `<Block/>` Component
+
+The `Block` component is a helper component that lets you write these common less-reusable elements more succinctly inline.
+
+The above example might turn into:
+
+```js
+<Block marginSides="scale200" marginEnds="scale150" color="grey200" display="flex" flexDirection="column">
```

Comment:
should `scale200` and these values rather came from the `theme` object? or that was the plane, you just wanted to keep the example simple?

Reply:
The idea was that internally `Block` would first check if the a prop is a theme constant (like `scale200`), and if so it would pull it from the theme object. But in theory a user could also pass it from theme themselves:

```js
import {LightTheme} from 'baseui';
export default () => <Block marginSides={$theme.sizing.scale200} />;
```

just requires importing and a few more keystrokes to access the constant from theme yourself. I'm sort of indifferent to which approach is used.